### PR TITLE
If github status URL returns a 404, bypass check and print an error i…

### PR DESCRIPTION
**Pull request by jaraco**
…nstead. Ref #65

### CODE REVIEW

**Comment by lazka** [comment]
I've just had the same issue with verify_github_issue_import_finished raising 404. The issue was imported properly so I'd suggest retrying with a timeout.

**Comment by jaraco** [comment]
Retrying isn't suitable because that check URL will forever be 404... And since the issue imported, a retry to import is not appropriate either. I think the only option with the 404 is to ignore the error... Until Github can determine why it's issuing invalid URLs.

On Apr 7, 2016, at 20:18, Christoph Reiter <notifications@github.com<mailto:notifications@github.com>> wrote:


I've just had the same issue with verify_github_issue_import_finished raising 404. The issue was imported properly so I'd suggest retrying with a timeout.

**Comment by jeffwidman** [comment]
> Retrying isn't suitable because that check URL will forever be 404... And since the issue imported, a retry to import is not appropriate either.

Is that really true?
My memory is a little hazy from a year ago, but I thought the check url sometimes stopped 404'ing after a period of time and sometimes didn't.

**Comment by jaraco** [comment]
> Is that really true?

I know that at least for some cases, it remained 404 for several minutes, even after a successful import. It doesn't seem unreasonable to retry, especially if supplying a cache-busting header might help. Still, adding a retry would be an enhancement to this proposed behavior, which it itself superior to the status quo.

File: migrate.py [hunk 1]
```diff
@@ -186,13 +186,16 @@ def main(options):
...
-                status_url, options.gh_auth, headers).json()['issue_url']
+            resp = verify_github_issue_import_finished(
+                status_url, options.gh_auth, headers)
+
             # verify GH & BB issue IDs match
             # if this fails, convert_links() will have incorrect output
             # this will fail if the GH repository has pre-existing issues
-            gh_issue_id = int(gh_issue_url.split('/')[-1])
-            assert gh_issue_id == issue['local_id']
+            if resp:
```
Comment (jeffwidman) [inline migrate.py]: Could use a comment right here explaining why we don't assume a response... something as simple as "Github occasionally inexplicably returns 404 even when successfully importing an issue. For details see #77"

File: migrate.py [hunk 2]
```diff
@@ -459,6 +462,9 @@ def verify_github_issue_import_finished(status_url, auth, headers):
     """
     while True:  # keep checking until status is something other than 'pending'
         respo = requests.get(status_url, auth=auth, headers=headers)
+        if respo.status_code == 404:
+            print("404 retrieving status URL", status_url)
```
Comment (jeffwidman) [inline migrate.py]: Also expand this message slightly so the user understands things might not have completely blown up. Again, something like "GitHub sometimes inexplicably returns a 404 for the check url for a single issue. Sometimes these issues still import successfully, sometimes not. For details, see #77."

**Review by jeffwidman — COMMENTED** [review]
Looking at this now, it makes a lot of sense to me, and I don't know why I was hesitant to merge it earlier. Sorry about that!

If you don't mind rebasing and adding a couple of comments, then I'll happily merge.

**Comment by jaraco** [comment]
So what you have here in this latest push is somewhat different. In ad5f582, you have the original bypass commit. In 330beb1, you have the requested additional hint and reference to this PR. In e32753f, you have an additional change which I added later, as I apparently found that a 403 will sometimes occur as well (probably due to rate-limiting, which I address in another branch). In fed22b9, I merge the two new commits so that if you want to merge just one of those but not the other, you can do so readily (or I can update the head for this branch), though I recommend pulling it all together.

**Comment by jeffwidman** [comment]
Looks good, let's ship it!

**Pull request merged**